### ISSUE TITLE
Bug: Type incompatibility with ESLint types (from @vitest/eslint-plugin)

### ISSUE THREAD

**Issue by danielrentz** [typescript-eslint/typescript-eslint#10597]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

utils

### Repro Code

```TypeScript
import type { Linter } from "eslint";
import vitest from "@vitest/eslint-plugin";
const TEST: Linter.Config = { plugins: { vitest } };
```

### Expected Result

no type error

### Actual Result

type error

### Additional Info

After upgrading `eslint`, `typescript-eslint`, and `@vitest/eslint-plugin`, linting fails with a type error.

```
src/eslint/env/vitest.ts:40:17 - error TS2322: Type 'VitestPlugin' is not assignable to type 'Plugin'.
  Types of property 'processors' are incompatible.
    Type 'Record<string, ProcessorModule> | undefined' is not assignable to type 'Record<string, Processor<string | ProcessorFile>> | undefined'.
      Type 'Record<string, ProcessorModule>' is not assignable to type 'Record<string, Processor<string | ProcessorFile>>'.
        'string' index signatures are incompatible.
          Type 'ProcessorModule' is not assignable to type 'Processor<string | ProcessorFile>'.
            Types of property 'postprocess' are incompatible.
              Type 'PostProcess | undefined' is not assignable to type '((messages: LintMessage[][], filename: string) => LintMessage[]) | undefined'.
                Type 'PostProcess' is not assignable to type '(messages: LintMessage[][], filename: string) => LintMessage[]'.
                  Types of parameters 'messagesList' and 'messages' are incompatible.
                    Type 'import(".../node_modules/eslint/lib/types/index").Linter.LintMessage[][]' is not assignable to type 'import(".../node_modules/@typescript-eslint/utils/dist/ts-eslint/Linter").Linter.LintMessage[][]'.
                      Type 'import(".../node_modules/eslint/lib/types/index").Linter.LintMessage[]' is not assignable to type 'import(".../node_modules/@typescript-eslint/utils/dist/ts-eslint/Linter").Linter.LintMessage[]'.
                        Property 'source' is missing in type 'import(".../node_modules/eslint/lib/types/index").Linter.LintMessage' but required in type 'import(".../node_modules/@typescript-eslint/utils/dist/ts-eslint/Linter").Linter.LintMessage'.

40                 vitest: vitestPlugin,
                   ~~~~~~

  ../../node_modules/@typescript-eslint/utils/dist/ts-eslint/Linter.d.ts:183:9
    183         source: string | null;
                ~~~~~~
    'source' is declared here.
```

Root cause:

```
Property 'source' is missing in type 'import(".../node_modules/eslint/lib/types/index").Linter.LintMessage'
but required in type 'import(".../node_modules/@typescript-eslint/utils/dist/ts-eslint/Linter").Linter.LintMessage'.
```

### Versions

| package | version |
| - | - |
| `@typescript-eslint/utils` | `8.19.0` |
| `TypeScript` | `5.7.2` |
| `ESLint` | `9.17.0` |
| `@vitest/eslint-plugin` | `1.1.23` |

**Comment by ronami** [comment]
I can reproduce the issue, but seems like it's been resolved a couple of days ago on [eslint-plugin-vitest v1.1.24](https://github.com/vitest-dev/eslint-plugin-vitest/releases/tag/v1.1.24).

**Comment by danielrentz** [comment]
Right... But the type inconsistency in this repo remains I think.
